application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java

```diff
@@ -137,7 +138,8 @@ protected <T extends HousekeeperTask> void processTask(ToHousekeeperServiceMsg m
             if (e instanceof ExecutionException) {
                 error = e.getCause();
             } else if (e instanceof TimeoutException) {
-                error = new TimeoutException("Timeout after " + config.getTaskProcessingTimeout() + " seconds");
+                future.cancel(true); // interrupting the task
+                error = new TimeoutException("Timeout after " + config.getTaskProcessingTimeout() + " ms");
```

Comment:
I suggest to add to the text a keyword for easy full test search or grep, like 
`Housekeeper timeout ...`
and maybe some basic detail - tenant id, task details

Reply:
Everything is logged already:
![image](https://github.com/user-attachments/assets/ab30a673-0a56-4e8a-ba76-ad768858d14a)